index.js

```diff
@@ -48,12 +49,42 @@ module.exports = {
       ssb.metafeeds.findOrCreate(details, cb)
     }
 
-    function findOrCreateGroupFeed(input, cb) {
-      const secret = Buffer.isBuffer(input) ? new SecretKey(input) : input
+    function findEmptyGroupFeed(rootId, cb) {
+      console.log('in empty function')
+      pull(
+        ssb.metafeeds.branchStream({ root: rootId, live: false }),
+        pull.filter((branch) => branch.length === 4),
+        pull.map((branch) => branch[3]),
+        pull.filter((feed) => feed.recps),
+        paraMap((feed, cb) => {
+          return pull(
+            ssb.db.query(where(author(feed.id)), toPullStream()),
+            pull.take(1),
+            pull.collect((err, res) => {
+              if (res.length === 0) {
+                // we're only interested in empty feeds
+                return cb(null, feed)
+              } else {
+                return cb()
+              }
+            })
+          )
+        }),
+        pull.filter(),
+        pull.unique('id'),
+        //pull.flatten(),
+        pull.collect((err, empties) => {
+          if (err) return cb(err)
+          return cb(null, empties[0])
+        })
```

Comment:
This pull.collect could be replaced with `pull.take(1)` and `pull.drain`, because pull.collect can be bad for memory consumption.

Reply:
done